inc/saiport.h

```diff
@@ -174,13 +174,34 @@ typedef enum _sai_port_fec_mode_t
     /** No FEC */
     SAI_PORT_FEC_MODE_NONE,
 
-    /** Enable RS-FEC - 25G, 50G, 100G ports */
+    /** Enable RS-FEC - 25G, 50G, 100G ports. The specific RS-FEC mode will be automatically determined. */
     SAI_PORT_FEC_MODE_RS,
 
     /** Enable FC-FEC - 10G, 25G, 40G, 50G ports */
     SAI_PORT_FEC_MODE_FC,
 } sai_port_fec_mode_t;
 
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_FEC_MODE_EXTENDED
+ */
+typedef enum _sai_port_fec_mode_extended_t
```

Comment:
RS-FEC has 2 variants - RS(514,528) for NRZ and RS(514,544) for PAM4. As per IEEE802.3 the 514 value is used to calculate the number of parity symbols carried to protect the data so no need to mention it in specific.

Reply:
Thanks, so we can skip the 514. CL74 is FC FEC, so that one doesn't need to be added. I'm not finding good details on CL108's Reed-Solomon length/symbols parameters, so I'm reluctant to add it unless it's done as RS-CL108 (which doesn't match the naming of the other FEC types).

Reply:
I think we can also skip CL108. As per IEEE802.3 the 25GBASE-R RS-FEC sublayer employs RS(528,514).

Reply:
Ack, thanks for helping out with this!